modules/auxiliary/gather/elasticsearch_enum.rb

```diff
@@ -0,0 +1,400 @@
+
+    res = send_request_cgi(request)
+
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::NoAccess, 'Credentials required, or incorrect') if res.code == 401
+
+    # leaving this here for future travelers, this header was added in 7.14.0 https://www.elastic.co/guide/en/elasticsearch/reference/7.17/release-notes-7.14.0.html
+    # so it isn't too reliable to check for
+    # fail_with(Failure::Unreachable, "#{peer} - Elasticsearch not detected in X-elastic-product header") unless res.headers['X-elastic-product'] == 'Elasticsearch'
+
+    if res.code == 200 && !res.body.empty?
+      json_body = res.get_json_document
+      if json_body.empty?
+        vprint_error('Unable to parse JSON')
+        return
+      end
+    end
+
+    fail_with(Failure::Unreachable, "#{peer} - Elasticsearch cluster name not found, likely not Elasticsearch server") unless json_body['cluster_name']
+
+    elastic_table = Rex::Text::Table.new(
+      'Header' => 'Elastic Information',
+      'Indent' => 2,
+      'Columns' =>
+      [
+        'Name',
+        'Cluster Name',
+        'Version',
+        'Build Type',
+        'Lucene Version'
+      ]
+    )
+
+    elastic_table << [
+      json_body['name'],
+      json_body['cluster_name'],
+      json_body.dig('version', 'number'),
+      json_body.dig('version', 'build_type'),
+      json_body.dig('version', 'lucene_version'),
+    ]
+    print_good(elastic_table.to_s)
+
+    @service = report_service(
+      host: rhost,
+      port: rport,
+      proto: 'tcp',
+      name: 'elasticsearch'
+    )
+  end
+
+  def get_users
+    vprint_status('Querying user information...')
+    request = {
+      'uri' => normalize_uri(target_uri.path, '_security', 'user/'),
+      'method' => 'GET'
+    }
+    request['authorization'] = basic_auth(datastore['USERNAME'], datastore['PASSWORD']) if datastore['USERNAME'] || datastore['PASSWORD']
+
+    res = send_request_cgi(request)
+
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::NoAccess, 'Credentials required, or incorrect') if res.code == 401
+
+    if res.code == 200 && !res.body.empty?
+      json_body = res.get_json_document
+      if json_body.empty?
+        vprint_error('Unable to parse JSON')
+        return
+      end
+    end
+
+    elastic_table = Rex::Text::Table.new(
+      'Header' => 'User Information',
+      'Indent' => 2,
+      'Columns' =>
+      [
+        'Name',
+        'Roles',
+        'Email',
+        'Metadata',
+        'Enabled'
+      ]
+    )
+    json_body.each do |username, attributes|
+      elastic_table << [
+        username,
+        attributes['roles'],
+        attributes['email'],
+        attributes['metadata'],
+        attributes['enabled'],
+      ]
+    end
+    print_good(elastic_table.to_s)
```

Comment:
I'm getting another exception here. I don't think I have any users setup.

```
[-] Auxiliary failed: NoMethodError undefined method `each' for nil:NilClass
[-] Call stack:
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/gather/elasticsearch_enum.rb:375:in `get_users'
[-]   /home/smcintyre/Repositories/metasploit-framework.pr/modules/auxiliary/gather/elasticsearch_enum.rb:399:in `run'
[*] Auxiliary module execution completed
msf6 auxiliary(gather/elasticsearch_enum) > 
```

Reply:
more likely the user you gave didn't have permission to view that data, which is a good case to test